src/components/TextInput/TextInput.tsx

```diff
@@ -147,9 +155,14 @@ export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(funct
                 <InputControl {...props} {...commonProps} controlRef={handleRef} />
             )}
 
-            {isErrorMsgVisible && <div className={b('error')}>{error}</div>}
+            {isErrorMsgVisible && (
+                <div data-qa={innerQa('error')} className={b('error')}>
+                    {error}
+                </div>
+            )}
             {hasClear && (
                 <span
+                    data-qa={innerQa('clear')}
```

Comment:
I'm not sure about using `data-qa` for component's internals. `data-qa` attribute was designed to be on the root element only

Reply:
ok, how can I find this element without data-qa attribute? by class?

Reply:
by class or by tag